src/cr/cube/dimension.py

```diff
@@ -725,6 +734,214 @@ def _elements(self):
         return tuple(element for element in self._all_elements if not element.missing)
 
 
+class _DimensionShimElementIds(object):
+    """Object used to replace element ids with alias for subvariables
+
+    We want to move to a world where elements on a subvariables dimension are
+    identified by their alias, but right now the "element_id" from zz9 is
+    an index, and the transforms have several different ways to refer to
+    subvariables.
+
+    Types of identifiers for subvariables (and derived insertions):
+    - "element_id": Stored in the cube result as the object name in
+        `dimensions[i].type.elements[j].id`. For subvariables, zz9 currently puts
+        the index integer here. Long term zz9 may change this to the the alias.
+    - "subvariable_id": Subvariables have an id stored in
+        `dimensions[i].type.elements[j].value.id`, generally this is a 4 digit,
+        0-padded index of the subvariable when it was first created (eg "0001",
+        "0002", ...), though it is not required to be. For derived insertions,
+        currently the name is used here.
+    - "alias": Subvariables also have an alias that identifies them. It
+    """
```

Comment:
Couple formatting changes.

* We put a period after the summary-line of the docstring. It's not always a whole sentence, but it's just what we've come to do.
* There should be a blank like before the first bullet. The text in a docstring is restructuredText interpreted by Sphinx to generate the documentation and it won't know what to do without a space before the bulleted list.
* The default first-level bullet character for restructuredText is an asterisk ("*"), so probably doesn't hurt to stick with that for consistency.
* Second and later lines of a bullet-point should line up under the first character of the first line of the bullet point, like:

  ```rst
  - "element_id": Stored in the cube result as the object name in
    `dimensions[i].type.elements[j].id`. For subvariables, zz9 currently puts
    the index integer here. Long term zz9 may change this to the the alias.
  ```

  I'm not sure how it gets formatted if it doesn't align, but I'm pretty sure it's not what we would expect or want.